### DISCUSSION TITLE
Upload Limitations

### DISCUSSION THREAD

**Discussion by markdrrr** [pillargg/youtube-upload#12] [Q&A] [answered]
Is there any limit on the number of publications per day? 
______
How to add thumbnail not a link but a file?

**Comment by chand1012** [comment, marked as answer]
> 
> 
> Is there any limit on the number of publications per day?
> 
> How to add thumbnail not a link but a file?

The upload limit is imposed by Google, not us. See [here](https://developers.google.com/youtube/v3/getting-started#quota).

We plan on adding functionality for thumbnail uploading in the future, but for now it is via a URL. In theory you could upload the file to some sort of file server, either an S3 bucket or another type of scriptable image upload service (Imgur wouldn't be a bad option), and then get the public URL of the file for the function call.

**Comment by chen-zhuofu** [comment]
I noticed two difficulties so far to apply the binary.

1. The video that I upload will always be marked as private, while I need them to be public. 
2. There is a 10000 daily quota limit on Youtube API(Which only allows uploading 3~5 videos per day). How we can get rid of this limit?

Thanks in advance

**Reply by chand1012** [reply]
You have to get both of those restrictions to be lifted by Google. See [here](https://developers.google.com/youtube/v3/docs/videos?hl=en). TLDR, via the linked article. 

>All videos uploaded via the [`videos.insert`](https://developers.google.com/youtube/v3/docs/videos/insert) endpoint from unverified API projects created after 28 July 2020 will be restricted to private viewing mode. To lift this restriction, each API project must [undergo an audit](https://support.google.com/youtube/contact/yt_api_form) to verify compliance with the [Terms of Service](https://developers.google.com/youtube/terms/api-services-terms-of-service). Please see the [API Revision History](https://developers.google.com/youtube/v3/revision_history#release_notes_07_28_2020) for more details.

**Reply by chen-zhuofu** [reply]
Thanks for the info. I submitted two applications for the quota and audit. But looks like it will take very long to know the result. Without that two approval, I guess the binary becomes pretty much useless. By any chance do you know if there is another tool or existing project that has already passed the Google audit or acquired more quota(>1,000,000 for example)?

**Reply by chand1012** [reply]
> By any chance do you know if there is another tool or existing project that has already passed the Google audit or acquired more quota(>1,000,000 for example)?

Its not the library that Google limits, but the application you are trying to do that Google audits. The tools or libraries you use make no difference as far as Google is concerned, they just don't want you to make an application that will abuse their APIs.

**Reply by chen-zhuofu** [reply]
make sense
